gradle/render-javadoc.gradle

```diff
@@ -15,93 +15,105 @@
-            includenosourcepackages: "true",
-            author: "true",
-            version: "true",
-            linksource: linksource,
-            use: "true",
-            failonerror: "true",
-            locale: "en_US",
-            windowtitle: title,
-            doctitle: title,
-            maxmemory: "512m",
-            classpath: sourceSets.main.compileClasspath.asPath,
-            bottom: "<i>Copyright &copy; 2000-${buildYear} Apache Software Foundation. All Rights Reserved.</i>"
-        ) {
-          srcDirs.collect { srcDir ->
-            packageset(dir: srcDir)
+        project.exec {
+          executable javadocCmd
+
+          args += [ "-overview", file("src/java/overview.html").toString() ]
+          args += [ "-sourcepath", srcDirs.join(" ") ]
+          args += [ "-subpackages", project.path.startsWith(":lucene") ? "org.apache.lucene" : "org.apache.solr"]
+          args += [ "-d", project.javadoc.destinationDir.toString() ]
+          args += [ "-protected" ]
+          args += [ "-encoding", "UTF-8" ]
+          args += [ "-charset", "UTF-8" ]
+          args += [ "-docencoding", "UTF-8" ]
+          args += [ "-noindex" ]
+          args += [ "-author" ]
+          args += [ "-version" ]
+          if (linksource) {
+            args += [ "-linksource" ]
           }
+          args += [ "-use" ]
+          args += [ "-locale", "en_US" ]
+          args += [ "-windowtitle", title ]
+          args += [ "-doctitle", title ]
+          args += [ "-classpath", sourceSets.main.compileClasspath.asPath ]
+          args += [ "-bottom", "<i>Copyright &copy; 2000-${buildYear} Apache Software Foundation. All Rights Reserved.</i>" ]
 
-          tag(name: "lucene.experimental", description: "WARNING: This API is experimental and might change in incompatible ways in the next release.")
-          tag(name: "lucene.internal", description: "NOTE: This API is for internal purposes only and might change in incompatible ways in the next release.")
-          tag(name: "lucene.spi", description: "SPI Name (Note: This is case-insensitive. e.g., if the name is 'htmlStrip', 'htmlstrip' can be used when looking up the service):", scope: "types")
+          args += [ "-tag", "lucene.experimental:a:WARNING: This API is experimental and might change in incompatible ways in the next release." ]
+          args += [ "-tag", "lucene.internal:a:NOTE: This API is for internal purposes only and might change in incompatible ways in the next release." ]
+          args += [ "-tag", "lucene.spi:t:SPI Name (case-insensitive: if the name is 'htmlStrip', 'htmlstrip' can be used when looking up the service)." ]
 
           // resolve links to JavaSE and JUnit API
-          link(offline: "true", href: "https://docs.oracle.com/en/java/javase/11/docs/api/", packageListLoc: project(":lucene").file("tools/javadoc/java11/").toString())
+          args += [ "-linkoffline", javaSEDocUrl, project(":lucene").file("tools/javadoc/java11/").toURL() ]
```

Comment:
Why is toURL required? It was a path previously.

Reply:
According to -linkoffline documentation the second parameter takes url, so I used toURL() here. But file path is also fine (there is no difference in terms of generated documents). 
https://docs.oracle.com/en/java/javase/11/tools/javadoc.html
```
-linkoffline url1 url2
```

Reply:
This documentation looks weird to me ;) Specifically the "relative" url2 is not clear. If you do toURL() I think it'll be converted into a full file://.../ URL and it's no longer relative... There is also the question of what it does on Windows where file paths are quite odd.

Reply:
I changed toURL() to toString(). Both methods return absolute url / path and equally effective for me (MacOS and Linux) here, but I cannot confirm the outputs on Windows file system soon...

Reply:
I can check it on Windows. Let me know what to run, specifically, and if you need anything from me.

Reply:
Thanks, i'd be glad if you check the (absolute) links are correctly created by changes here when built on Windows. :)
```
./gradlew :lucene:analysis:common:renderJavadoc
```

This should generate hyperlinks to core module, for example 
<img width="620" alt="スクリーンショット 2020-03-31 18 36 01" src="https://user-images.githubusercontent.com/1825333/78011472-97ea2700-737e-11ea-89d5-eda9197fdf9f.png">
Here, `org.apache.lucene.analysis.Analyzer` should be `<a>` link to 
`https://lucene.apache.org/core/9_0_0-SNAPSHOT/core/org/apache/lucene/analysis/Analyzer.html` (url will be changed depending on project.version).

Reply:
Darn. I get execution error:

    > Task :lucene:core:renderJavadoc FAILED
    javadoc: error - Illegal package name: "<i>Copyright &copy; 2000-2020 Apache Software Foundation. All Rights Reserved.</i>"
    1 error

    FAILURE: Build failed with an exception.

    * Where:
    Script 'O:\repos\lucene-gradle-master\gradle\render-javadoc.gradle' line: 57

    * What went wrong:
    Execution failed for task ':lucene:core:renderJavadoc'.
    > Process 'command 'C:\Tools\java\jdk\bin\javadoc.exe'' finished with non-zero exit value 2

    * Try:
    Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

    * Get more help at https://help.gradle.org

    BUILD FAILED in 15s
    2 actionable tasks: 1 executed, 1 up-to-date

I think it's because something is escaped improperly. Let me look into this.

Reply:
I think the generation is working properly:
```
<li><a href="https://lucene.apache.org/core/9_0_0-SNAPSHOT/core/org/apache/lucene/analysis/Analyzer.html?is-external=true" title="class or interface in org.apache.lucene.analysis" class="externalLink">org.apache.lucene.analysis.Analyzer</a></li>
```

I am surprised this worked for you since the classpath for one of the modules is empty and this created an empty argument (which was skipped). Hence the problem above. I had to add:

          if (!sourceSets.main.compileClasspath.isEmpty()) {
            args += [ "-classpath", sourceSets.main.compileClasspath.asPath ]
          } 

to make it work.

I am also inclined to generate those arguments into a temporary file and use javadoc's "@file" syntax to pass arguments from that file. This is what gradle does; it also makes inspecting exact options passed to javadoc a tad easier (the file is task-temporary but remains under tmp if the task fails).

Reply:
I fixed this on https://github.com/apache/lucene-solr/pull/1388/commits/f48c6ecbe3b5a47ae2cc677dae8e1c44b513ac9c.

> I am also inclined to generate those arguments into a temporary file and use javadoc's "@file" syntax to pass arguments from that file. This is what gradle does; it also makes inspecting exact options passed to javadoc a tad easier (the file is task-temporary but remains under tmp if the task fails).

Ok, I can do this. Let me know another PR would be preferred for the change.

Reply:
Either way, whatever is more convenient for you. :)

Reply:
hi @dweiss,
I made following changes to the PR:
- save javadoc options to a file under the tmp directory for each task (_project_/build/tmp/renderJavadoc).
- pass the options file to javadoc tool via "@file" argument.
- when the task fails, it shows raw javadoc tool err message and options file path for debugging.

I've tested it on MacOS/Fedora Linux/Windows10. Works for me, but I'd appreciate if you could cross-check especially if you come up with some edge cases on Windows.